src/views/charts/TradingRewardsChart.tsx

```diff
@@ -0,0 +1,261 @@
+      }
+    },
+    [now, rewardsHistoryStartDate]
+  );
+
+  // Unselect selected period in toggle if user zooms in/out
+  const onZoomSnap = useMemo(
+    () =>
+      debounce(({ zoomDomain }: { zoomDomain?: number }) => {
+        if (zoomDomain) {
+          setIsZooming(!tradingRewardsPeriods.map(msForPeriod).includes(zoomDomain));
+        }
+      }, 200),
+    [msForPeriod]
+  );
+
+  const onToggleInteract = () => setIsZooming(false);
+
+  const rewardsData = useMemo(
+    () =>
+      periodTradingRewards && canViewAccount
+        ? periodTradingRewards
+            .toArray()
+            .reverse()
+            .map(
+              (datum) =>
+                ({
+                  date: new Date(datum.endedAtInMilliseconds).valueOf(),
+                  cumulativeAmount: datum.cumulativeAmount,
+                } as TradingRewardsDatum)
+            )
+        : [],
+    [periodTradingRewards, canViewAccount]
+  );
+
+  const series = useMemo(
+    () => [
+      {
+        dataKey: 'trading-rewards',
+        xAccessor: (datum: TradingRewardsDatum) => datum?.date,
+        yAccessor: (datum: TradingRewardsDatum) => datum?.cumulativeAmount,
+        colorAccessor: () => 'var(--trading-rewards-line-color)',
+        getCurve: () => curveLinear,
+      },
+    ],
+    []
+  );
+
+  const language = navigator.language || 'en-US';
+
+  const tickFormatY = useCallback(
+    (value: number) =>
+      Intl.NumberFormat(language, {
+        style: 'decimal',
+        notation: 'compact',
+        maximumSignificantDigits: 3,
+      })
+        .format(value)
+        .toLowerCase(),
+    [language]
+  );
+
+  const renderTooltip = useCallback(() => <div />, []);
+  const defaultZoomDomain = isZooming ? undefined : msForPeriod(selectedPeriod);
+
+  const toggleGroupItems = useMemo(() => {
+    return tradingRewardsPeriods.map((period: TradingRewardsPeriod) => ({
+      value: period,
+      label:
+        period === TradingRewardsPeriod.PeriodAllTime
+          ? stringGetter({ key: STRING_KEYS.ALL })
+          : formatRelativeTime(msForPeriod(period), {
+              locale: selectedLocale,
+              relativeToTimestamp: 0,
+              largestUnit: 'day',
+            }),
+    }));
+  }, [stringGetter, msForPeriod, selectedLocale]);
+
+  const setTradingRewardsPeriod = useCallback(
+    (value: string) => setSelectedPeriod(value as TradingRewardsPeriod),
+    []
+  );
+
+  return (
+    <>
+      {rewardsData.length === 0 ? undefined : (
+        <$TitleContainer>
+          <$Title>
+            {stringGetter({
+              key: STRING_KEYS.TRADING_REWARDS,
+            })}
+          </$Title>
+
+          <ToggleGroup
+            items={toggleGroupItems}
+            value={isZooming ? '' : selectedPeriod}
+            onValueChange={setTradingRewardsPeriod}
+            onInteraction={onToggleInteract}
+          />
+        </$TitleContainer>
+      )}
+      <TimeSeriesChart
```

Comment:
nit: it seems like the right number is cutoff, maybe more padding?
<img width="272" alt="Screenshot 2024-05-28 at 4 26 23 PM" src="https://github.com/dydxprotocol/v4-web/assets/9400120/24553fe6-8e5e-4745-9cb8-69b9765f01c0">

Reply:
hmm yeah unsure about the balance here - number is always truncated to 3 sig digs at most, but that can get long depending on how many 0s you have before that. I was optimizing for most people having >1 token as a reward (so we'll have 3 digits rendered at most); open to bumping it though